pkg/scheduler/framework/plugins/podtopologyspread/scoring_test.go

```diff
@@ -427,6 +433,40 @@ func TestPodTopologySpreadScore(t *testing.T) {
 				{Name: "node-d", Score: 100},
 			},
 		},
+		{
+			name: "system defaulting, nodes don't have zone, pods match service",
+			pod:  st.MakePod().Name("p").Label("foo", "").Obj(),
+			existingPods: []*v1.Pod{
+				// matching pods spread as 4/3/2/1.
+				st.MakePod().Name("p-a1").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-a2").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-a3").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-a4").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b1").Node("node-b").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b2").Node("node-b").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b3").Node("node-b").Label("foo", "").Obj(),
+				st.MakePod().Name("p-c1").Node("node-c").Label("foo", "").Obj(),
+				st.MakePod().Name("p-c2").Node("node-c").Label("foo", "").Obj(),
+				st.MakePod().Name("p-d1").Node("node-d").Label("foo", "").Obj(),
+			},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label(v1.LabelHostname, "node-a").Obj(),
+				st.MakeNode().Name("node-b").Label(v1.LabelHostname, "node-b").Obj(),
+				st.MakeNode().Name("node-c").Label(v1.LabelHostname, "node-c").Obj(),
+				st.MakeNode().Name("node-d").Label(v1.LabelHostname, "node-d").Obj(),
```

Comment:
For curiosity: in addition to zone label, should we consider the case that all or partial nodes don't carry standard host label?

Reply:
Is that even possible? It seems that kubelet takes care of it https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetesiohostname

Interestingly, given that we special-case hostname to do the count in `Score` (instead of `PreScore`), we could make it work. However, I don't think we need to support that.

Reply:
> Is that even possible?

I was just curious. Personally I'm with you to believe it's sane defaults for every cluster.